command/server/service.go

```diff
@@ -107,3 +109,37 @@ func (s *Server) ChainSetHead(ctx context.Context, req *proto.ChainSetHeadReques
 	s.backend.APIBackend.SetHead(req.Number)
 	return &proto.ChainSetHeadResponse{}, nil
 }
+
+func ConvertBlockToBlockStub(blocks []*types.Block) []*proto.BlockStub {
+
+	var blockStubs []*proto.BlockStub
+
+	for _, block := range blocks {
+		blockStub := &proto.BlockStub{
+			Hash:   block.Hash().String(),
+			Number: block.NumberU64(),
+		}
+		blockStubs = append(blockStubs, blockStub)
+	}
+
+	return blockStubs
+}
+
+func (s *Server) ChainWatch(req *proto.ChainWatchRequest, reply proto.Bor_ChainWatchServer) error {
+	// 1. start the feed to the blockchain events
+	// 2. for each event send a proto.ChainWatchResponse
+
+	chain2HeadChanSize := 10
+
+	chain2HeadCh := make(chan core.Chain2HeadEvent, chain2HeadChanSize)
+	s.headSub = s.backend.APIBackend.SubscribeChain2HeadEvent(chain2HeadCh)
```

Comment:
This is not correct, each ```ChainWatch``` call should be independent from each other so the subscription reference should local to the call. Remove the ```headSub``` variable from Server too.

Reply:
You should also close the subscription. Do:
```
defer headSub.Unsubscribe()
```
to unsubscribe the subscription once the function is over.